### ISSUE TITLE
v4.2.2 and v4.3.0 tags

### ISSUE THREAD

**Issue by 12wrigja** [GoogleChromeLabs/jsbi#92]
Whilst following the instructions in the README, I accidentally pushed tags for both v4.2.2 and v4.3.0 (as I hadn't realized that the old tag wasn't deleted when I reset my working directory state back to upstream main).

I was able to cancel the publish workflow for v4.2.2 (but [v4.3.0](https://www.npmjs.com/package/jsbi/v/4.3.0) has been published).

Thoughts on how to reconcile this?

**Comment by 12wrigja** [comment]
My thoughts are to try and un-publish v4.3.0 and retry the publish for v4.2.2.

**Comment by 12wrigja** [comment]
On second thought, maybe we should merge the [v4.3.0 tagged commit](https://github.com/GoogleChromeLabs/jsbi/commit/c40371e089d1433bd8ff20b7a5be617d0561e85c) into main and publish both versions?

**Comment by 12wrigja** [comment]
After chatting with some colleagues about this situation, they suggested that:
 - It's possible to "archive" the 4.3.0 release on NPM and re-set the latest tag to 4.2.2, but would mean having to jump to 4.3.1 if we ever needed that. We'd also need to manually start the GitHub Actions workflow again to publish 4.2.2.
 - We could make this repo's state consistent with the 4.3.0 release by force-pushing the branch reference to line up with the relevant commit, and the only users affected should be anyone who is working on changes since that commit arrived in the main branch (probably nobody)? We'd probably need to also delete the 4.2.2 tag to avoid CI from re-publishing it.
 
I'm leaning towards the latter option, accepting that whilst the minor version bump isn't really correct use of semver for the changes involved it's not detrimental and makes updating later significantly less confusing.

@mathiasbynens I'd like to ask for your decision on which of these options to persue / help implementing.

I'm very surprised that the GitHub workflow doesn't seem to have the option to limit itself to "new tags published to the main branch", which would have prevented this. A [StackOverflow post](https://stackoverflow.com/questions/59874409/trigger-github-workflow-only-if-pushed-to-specific-branch-and-tag-exists) implies that it might be possible to check in the workflow whether the tag is on the main branch or not before continuing to publish which could be helpful.

**Comment by mathiasbynens** [comment]
The latter option SGTM.

> I'm very surprised that the GitHub workflow doesn't seem to have the option to limit itself to "new tags published to the main branch", which would have prevented this. A [StackOverflow post](https://stackoverflow.com/questions/59874409/trigger-github-workflow-only-if-pushed-to-specific-branch-and-tag-exists) implies that it might be possible to check in the workflow whether the tag is on the main branch or not before continuing to publish which could be helpful.

FWIW, the instructions at https://github.com/GoogleChromeLabs/jsbi#how-to-publish-a-new-release say to only ever create tags while on the `main` branch. As long as we stick to that, we’re fine. Enforcing it rather than relying on all maintainers to get it 100% of the time sounds good to me — I just hadn’t considered the need thus far.

**Comment by 12wrigja** [comment]
I did make the tags on the main branch in both cases, but when I reset my local working directory's state (to change from minor to patch version change), I didn't delete the tag that had been created. Technically, when I made the tag I was on the main branch, but later that branch reference was moved and when the tag was pushed it was no longer part of the main branch.

I've force-pushed the v4.3.0 commit to main, and it looks like all is well. It seems we don't need to delete the 4.2.2 tag, but maybe it's worth doing so for consistency/repo hygiene.
